src/v2/coins/abstractUtxoCoin.js

```diff
@@ -591,14 +619,14 @@ class AbstractUtxoCoin extends BaseCoin {
 
     addressDetails.coinSpecific.redeemScript = inputScript.toString('hex');
 
-    if ([this.constructor.AddressTypes.P2SH_P2WSH, this.constructor.AddressTypes.P2WSH].includes(addressType)) {
+    if ([Codes.UnspentTypeTcomb('p2shP2wsh'), Codes.UnspentTypeTcomb('p2wsh')].includes(addressType)) {
       const witnessScriptHash = bitcoin.crypto.sha256(inputScript);
       const redeemScript = bitcoin.script.witnessScriptHash.output.encode(witnessScriptHash);
       const redeemScriptHash = bitcoin.crypto.hash160(redeemScript);
       outputScript = bitcoin.script.scriptHash.output.encode(redeemScriptHash);
       addressDetails.coinSpecific.witnessScript = inputScript.toString('hex');
       addressDetails.coinSpecific.redeemScript = redeemScript.toString('hex');
-      if (addressType === this.constructor.AddressTypes.P2WSH) {
+      if (addressType === Codes.UnspentTypeTcomb('p2wsh')) {
```

Comment:
Maybe we should refactor to do the case differentiation before creating the redeemscript on the address in the first place and only create it when it's a wrapped segwit address.

Reply:
Done. Please re-review.